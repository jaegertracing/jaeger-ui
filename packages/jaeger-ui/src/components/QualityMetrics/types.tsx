// Copyright (c) 2020 Uber Technologies, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import * as React from 'react';

import { TExample } from '../common/ExamplesLink';
import { TColumnDef, TRow } from '../common/DetailsCard/types';

export type TQualityMetrics = {
  traceQualityDocumentationLink: string;
  bannerText?:
    | string
    | {
        value: string;
        styling: React.CSSProperties;
      };
  scores: {
    key: string;
    label: string;
    max: number;
    value: number;
  }[];
  metrics: {
    name: string;
    category: string; // matching some scores.key;
    description: string;
    metricDocumentationLink: string;
    metricWeight: number;
    passCount: number;
    passExamples?: TExample[];
    failureCount: number;
    failureExamples?: TExample[];
    exemptionCount?: number;
    exemptionExamples?: TExample[];
    details?: {
      description?: string;
      header?: string;
      columns: TColumnDef[];
      rows: TRow[];
    }[];
  }[];
  clients?: {
    version: string;
    minVersion: string;
    count: number;
    examples: TExample[];
  }[];
};
