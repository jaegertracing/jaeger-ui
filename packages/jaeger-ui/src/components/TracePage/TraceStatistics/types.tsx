// Copyright (c) 2020 The Jaeger Authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

export interface ITableSpan {
  type: string;
  name: string;
  count: number;
  total: number;
  avg: number;
  min: number;
  max: number;
  selfTotal: number;
  selfAvg: number;
  selfMin: number;
  selfMax: number;
  percent: number;
  isDetail: boolean;
  parentElement: string;
  color: string; // If it is a service name, the color will be set.
  searchColor: string;
  colorToPercent: string; // Color created by percent
}

export interface ITableValues {
  uid: string;
  value: number;
}

export interface IColumnValue {
  title: string;
  attribute: string;
  suffix: string;
  isDecimal: boolean;
}
