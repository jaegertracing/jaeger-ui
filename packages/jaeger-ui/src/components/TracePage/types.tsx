// Copyright (c) 2017 Uber Technologies, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import { TNil } from '../../types';

interface ITimeCursorUpdate {
  cursor: number | TNil;
}

interface ITimeReframeUpdate {
  reframe: {
    anchor: number;
    shift: number;
  };
}

interface ITimeShiftEndUpdate {
  shiftEnd: number;
}

interface ITimeShiftStartUpdate {
  shiftStart: number;
}

export type TUpdateViewRangeTimeFunction = (start: number, end: number, trackSrc?: string) => void;

export type ViewRangeTimeUpdate =
  | ITimeCursorUpdate
  | ITimeReframeUpdate
  | ITimeShiftEndUpdate
  | ITimeShiftStartUpdate;

export interface IViewRangeTime {
  current: [number, number];
  cursor?: number | TNil;
  reframe?: {
    anchor: number;
    shift: number;
  };
  shiftEnd?: number;
  shiftStart?: number;
}

export interface IViewRange {
  time: IViewRangeTime;
}
