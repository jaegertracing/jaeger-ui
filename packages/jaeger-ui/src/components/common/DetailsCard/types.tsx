// Copyright (c) 2020 Uber Technologies, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
import type React from 'react';
import { TExample } from '../ExamplesLink';

export type TStyledValue = {
  linkTo?: string;
  styling?: React.CSSProperties;
  value: string | React.ReactElement;
};

export type TColumnDef = {
  key: string;
  label?: string;
  preventSort?: boolean;
  styling?: React.CSSProperties;
};

export type TColumnDefs = (string | TColumnDef)[];

export type TRow = Record<string, TStyledValue | string | number | TExample[]>;

export type TDetails = string | string[] | TRow[];

export type TFilterDropdownProps = {
  clearFilters?: () => void;
  confirm: () => void;
  selectedKeys: React.Key[];
  setSelectedKeys: (selectedKeys: React.Key[]) => void;
};
