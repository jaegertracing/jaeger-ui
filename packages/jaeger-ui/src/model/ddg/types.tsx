// Copyright (c) 2019 Uber Technologies, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import { TVertex } from '@jaegertracing/plexus/lib/types';

import PathElem from './PathElem';

export { default as PathElem } from './PathElem';

export enum EViewModifier {
  None,
  Hovered,
  Selected,
  Emphasized = 1 << 2, // eslint-disable-line no-bitwise
  PathHovered = 1 << 3, // eslint-disable-line no-bitwise
}

export enum EDdgDensity {
  ExternalVsInternal = 'ext-vs-int',
  MostConcise = 'mc',
  OnePerLevel = 'per-level',
  PreventPathEntanglement = 'ppe',
  UpstreamVsDownstream = 'up-vs-down',
}

export enum ECheckedStatus {
  Empty = 'Empty',
  Full = 'Full',
  Partial = 'Partial',
}

export enum EDirection {
  Upstream = -1,
  Downstream = 1,
}

export type TDdgPayloadEntry = {
  operation: string;
  service: string;
};

export type TDdgPayloadPath = {
  path: TDdgPayloadEntry[];
  // TODO: Everett Tech Debt: Fix KeyValuePair types
  attributes: {
    key: 'exemplar_trace_id'; // eslint-disable-line camelcase
    value: string;
  }[];
};

export type TDdgPayload = {
  dependencies: TDdgPayloadPath[];
};

export type TDdgService = {
  name: string;
  operations: Map<string, TDdgOperation>;
};

export type TDdgOperation = {
  name: string;
  pathElems: PathElem[];
  service: TDdgService;
};

export type TDdgServiceMap = Map<string, TDdgService>;

export type TDdgPath = {
  focalIdx: number;
  members: PathElem[];
  traceIDs: string[];
};

export type TDdgDistanceToPathElems = Map<number, PathElem[]>;

export type TDdgModel = {
  distanceToPathElems: TDdgDistanceToPathElems;
  hash: string;
  paths: TDdgPath[];
  services: TDdgServiceMap;
  visIdxToPathElem: PathElem[];
};

export type TDdgVertex = TVertex<{
  isFocalNode: boolean;
  key: string;
  operation: string | string[] | null;
  service: string;
}>;

export type TDdgSparseUrlState = {
  density: EDdgDensity;
  decoration?: string;
  end?: number;
  hash?: string;
  operation?: string;
  service?: string;
  showOp?: boolean;
  start?: number;
  visEncoding?: string;
};

export type TDdgModelParams = {
  service: string;
  operation?: string;
  start: number;
  end: number;
};

export type TDdgActionMeta = {
  query: TDdgModelParams;
};

export type TDdgAddViewModifierPayload = TDdgModelParams & {
  // Number instead of EViewModifier so that multiple views can be changed at once.
  viewModifier: number;
  visibilityIndices: number[];
};

export type TDdgClearViewModifiersFromIndicesPayload = TDdgAddViewModifierPayload & { viewModifier?: void };

export type TDdgRemoveViewModifierFromIndicesPayload = TDdgAddViewModifierPayload;

export type TDdgRemoveViewModifierPayload = TDdgAddViewModifierPayload & { visibilityIndices?: void };

export type TDdgViewModifierRemovalPayload =
  | TDdgClearViewModifiersFromIndicesPayload
  | TDdgRemoveViewModifierFromIndicesPayload
  | TDdgRemoveViewModifierPayload;

export type THop = { distance: number; fullness: ECheckedStatus };
