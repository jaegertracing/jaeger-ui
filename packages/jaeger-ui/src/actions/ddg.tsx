// Copyright (c) 2017 Uber Technologies, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import _identity from 'lodash/identity';
import { createActions, ActionFunctionAny, Action } from 'redux-actions';

import {
  TDdgAddViewModifierPayload,
  TDdgClearViewModifiersFromIndicesPayload,
  TDdgRemoveViewModifierFromIndicesPayload,
  TDdgRemoveViewModifierPayload,
  TDdgViewModifierRemovalPayload,
} from '../model/ddg/types';
import generateActionTypes from '../utils/generate-action-types';

export const actionTypes = generateActionTypes('@jaeger-ui/DEEP-DEPENDENCY-GRAPH', [
  'ADD_VIEW_MODIFIER',
  'CLEAR_VIEW_MODIFIERS_FROM_INDICES',
  'REMOVE_VIEW_MODIFIER',
  'REMOVE_VIEW_MODIFIER_FROM_INDICES',
]);

const addViewModifier: (kwarg: TDdgAddViewModifierPayload) => TDdgAddViewModifierPayload = _identity;
const clearViewModifiersFromIndices: (
  kwarg: TDdgClearViewModifiersFromIndicesPayload
) => TDdgClearViewModifiersFromIndicesPayload = _identity;
const removeViewModifier: (kwarg: TDdgRemoveViewModifierPayload) => TDdgRemoveViewModifierPayload = _identity;
const removeViewModifierFromIndices: (
  kwarg: TDdgRemoveViewModifierFromIndicesPayload
) => TDdgRemoveViewModifierFromIndicesPayload = _identity;

const fullActions = createActions<TDdgAddViewModifierPayload | TDdgViewModifierRemovalPayload>({
  [actionTypes.ADD_VIEW_MODIFIER]: addViewModifier,
  [actionTypes.CLEAR_VIEW_MODIFIERS_FROM_INDICES]: clearViewModifiersFromIndices,
  [actionTypes.REMOVE_VIEW_MODIFIER]: removeViewModifier,
  [actionTypes.REMOVE_VIEW_MODIFIER_FROM_INDICES]: removeViewModifierFromIndices,
});

type FullActionsType = {
  jaegerUi: {
    deepDependencyGraph: Record<string, ActionFunctionAny<Action<TDdgViewModifierRemovalPayload>>>;
  };
};

export default (fullActions as unknown as FullActionsType).jaegerUi.deepDependencyGraph;
