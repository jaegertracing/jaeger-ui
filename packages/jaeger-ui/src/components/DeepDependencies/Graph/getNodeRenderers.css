/*
Copyright (c) 2019 Uber Technologies, Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

.DdgNode--VectorBorder {
  fill: #fff;
  stroke: #bbb;
  stroke-width: 2;
}

.DdgNode--VectorBorder.is-findMatch {
  /* This color may be overridden or removed when other coloring rationales are added */
  stroke: #fadb14;
}

.DdgNode--VectorBorder.is-focalNode {
  stroke: #eb2f96;
}

.DdgNode--VectorBorder.is-pathHovered,
.DdgNode--VectorBorder.is-hovered {
  stroke-opacity: 1;
}

.DdgNode--HtmlEmphasis {
  border-radius: 100%;
  bottom: -20px;
  content: '';
  left: -20px;
  position: absolute;
  right: -20px;
  top: -20px;
  z-index: -1;
  margin: 1px 0 0 1px;
}

.DdgNode--HtmlEmphasis.is-findMatch {
  background-image: repeating-linear-gradient(-60deg, #fffb8f, #fff566 3px, #fffb8f 5px);
  border: 1px dashed rgba(173, 139, 0, 0.5);
}

.DdgNode--HtmlEmphasis.is-focalNode {
  background-image: repeating-linear-gradient(30deg, transparent, #eb2f96 3px, transparent 5px);
  border: 1px dashed #eb2f96;
}

.DdgNode--HtmlEmphasis.is-focalNode.is-findMatch {
  background-image: repeating-linear-gradient(30deg, #fffb8f, #eb2f96 3px, #fffb8f 5px);
}

.DdgNode--VectorFindEmphasis--colorBand {
  stroke: #fef876;
  stroke-width: 10;
}
