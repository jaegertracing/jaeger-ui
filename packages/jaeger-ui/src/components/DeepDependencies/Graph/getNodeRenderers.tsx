// Copyright (c) 2019 Uber Technologies, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import * as React from 'react';
import cx from 'classnames';
import { TLayoutVertex } from '@jaegertracing/plexus/lib/types';

import { TDdgVertex, EViewModifier } from '../../../model/ddg/types';

import './getNodeRenderers.css';

export default function getNodeRenderers(findMatches: Set<string>, viewModifiers: Map<string, number>) {
  function vectorBorder(lv: TLayoutVertex<TDdgVertex>) {
    // eslint-disable-next-line no-bitwise
    const isHovered = (viewModifiers.get(lv.vertex.key) || 0) & EViewModifier.Hovered;
    // eslint-disable-next-line no-bitwise
    const isPathHovered = (viewModifiers.get(lv.vertex.key) || 0) & EViewModifier.PathHovered;
    const className = cx('DdgNode--VectorBorder', {
      'is-findMatch': findMatches.has(lv.vertex.key),
      'is-hovered': isHovered,
      'is-pathHovered': isPathHovered,
      'is-focalNode': lv.vertex.isFocalNode,
    });
    return (
      <circle
        className={className}
        vectorEffect="non-scaling-stroke"
        r={lv.width / 2 - 1}
        cx={lv.width / 2}
        cy={lv.width / 2}
      />
    );
  }

  function htmlEmphasis(lv: TLayoutVertex<{ isFocalNode: boolean }>) {
    const matchClasses = cx({
      'is-findMatch': findMatches.has(lv.vertex.key),
      'is-focalNode': lv.vertex.isFocalNode,
    });
    if (!matchClasses) {
      return null;
    }
    return <div className={`DdgNode--HtmlEmphasis ${matchClasses}`} />;
  }

  if (!findMatches.size) {
    return {
      vectorBorder,
      htmlEmphasis,
      vectorFindColorBand: null,
    };
  }

  function vectorFindColorBand(lv: TLayoutVertex) {
    if (!findMatches.has(lv.vertex.key)) {
      return null;
    }

    return (
      <circle
        className="DdgNode--VectorFindEmphasis--colorBand"
        vectorEffect="non-scaling-stroke"
        r={lv.width / 2 - 1}
        cx={lv.width / 2}
        cy={lv.width / 2}
      />
    );
  }

  return {
    htmlEmphasis,
    vectorBorder,
    vectorFindColorBand,
  };
}
