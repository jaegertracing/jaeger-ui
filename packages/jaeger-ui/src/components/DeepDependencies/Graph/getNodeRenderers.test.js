// Copyright (c) 2019 Uber Technologies, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import { shallow } from 'enzyme';

import getNodeRenderers from './getNodeRenderers';

import { EViewModifier } from '../../../model/ddg/types';

describe('getNodeRenderers', () => {
  const key = 'test vertex key';
  const lv = {
    vertex: {
      key,
    },
    height: 200,
    width: 100,
  };
  const focalLv = { ...lv, vertex: { key, isFocalNode: true } };

  describe('vectorBorder', () => {
    // Short, DRY way to calculate with (w/) versus (v) without (w/o)
    const wvwo = someBoolean => (someBoolean ? 'with' : 'without');

    [true, false].forEach(findMatch => {
      [true, false].forEach(hovered => {
        [true, false].forEach(pathHovered => {
          [true, false].forEach(focalNode => {
            it(`returns circle ${wvwo(findMatch)} .is-findMatch,\t${wvwo(hovered)} .is-hovered,\t${wvwo(
              pathHovered
            )} .is-pathHovered,\tand ${wvwo(focalNode)} .is-focalNode`, () => {
              const testLv = focalNode ? focalLv : lv;
              const findMatches = new Set(findMatch ? [testLv.vertex.key] : undefined);
              const vm =
                // eslint-disable-next-line no-bitwise
                (hovered ? EViewModifier.Hovered : 0) | (pathHovered ? EViewModifier.PathHovered : 0);
              const vms = new Map([[key, vm]]);
              expect(shallow(getNodeRenderers(findMatches, vms).vectorBorder(testLv))).toMatchSnapshot();
            });
          });
        });
      });
    });
  });

  describe('htmlEmphasis', () => {
    it('returns null if vertex is neither a findMatch nor focalNode', () => {
      expect(getNodeRenderers(new Set(), new Map()).htmlEmphasis(lv)).toBe(null);
    });

    it('returns div with .is-findMatch if vertex is a findMatch', () => {
      const wrapper = shallow(getNodeRenderers(new Set([lv.vertex.key]), new Map()).htmlEmphasis(lv));
      expect(wrapper.hasClass('is-findMatch')).toBe(true);
      expect(wrapper.type()).toBe('div');
    });

    it('returns div with .is-focalNode if vertex is a focalNode', () => {
      const wrapper = shallow(getNodeRenderers(new Set(), new Map()).htmlEmphasis(focalLv));
      expect(wrapper.hasClass('is-focalNode')).toBe(true);
      expect(wrapper.type()).toBe('div');
    });

    it('returns div with .is-findMatch and .is-focalNode if vertex is a focalNode and a findMatch', () => {
      const wrapper = shallow(
        getNodeRenderers(new Set([focalLv.vertex.key]), new Map()).htmlEmphasis(focalLv)
      );
      expect(wrapper.hasClass('is-findMatch')).toBe(true);
      expect(wrapper.hasClass('is-focalNode')).toBe(true);
      expect(wrapper.type()).toBe('div');
    });
  });

  describe('vectorFindColorBand', () => {
    it('is null if findMatches set is empty', () => {
      expect(getNodeRenderers(new Set(), new Map()).vectorFindColorBand).toBe(null);
    });

    it('returns null if provided vertex is not in set', () => {
      expect(
        getNodeRenderers(new Set([{ vertex: { key: `not-${key}` } }]), new Map()).vectorFindColorBand(lv)
      ).toBe(null);
    });

    it('returns circle with correct size and className', () => {
      expect(
        shallow(getNodeRenderers(new Set([lv.vertex.key]), new Map()).vectorFindColorBand(lv))
      ).toMatchSnapshot();
    });
  });
});
