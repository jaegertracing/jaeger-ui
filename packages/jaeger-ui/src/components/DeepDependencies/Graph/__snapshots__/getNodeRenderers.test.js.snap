// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getNodeRenderers vectorBorder returns circle with .is-findMatch,	with .is-hovered,	with .is-pathHovered,	and with .is-focalNode 1`] = `
<circle
  className="DdgNode--VectorBorder is-findMatch is-hovered is-pathHovered is-focalNode"
  cx={50}
  cy={50}
  r={49}
  vectorEffect="non-scaling-stroke"
/>
`;

exports[`getNodeRenderers vectorBorder returns circle with .is-findMatch,	with .is-hovered,	with .is-pathHovered,	and without .is-focalNode 1`] = `
<circle
  className="DdgNode--VectorBorder is-findMatch is-hovered is-pathHovered"
  cx={50}
  cy={50}
  r={49}
  vectorEffect="non-scaling-stroke"
/>
`;

exports[`getNodeRenderers vectorBorder returns circle with .is-findMatch,	with .is-hovered,	without .is-pathHovered,	and with .is-focalNode 1`] = `
<circle
  className="DdgNode--VectorBorder is-findMatch is-hovered is-focalNode"
  cx={50}
  cy={50}
  r={49}
  vectorEffect="non-scaling-stroke"
/>
`;

exports[`getNodeRenderers vectorBorder returns circle with .is-findMatch,	with .is-hovered,	without .is-pathHovered,	and without .is-focalNode 1`] = `
<circle
  className="DdgNode--VectorBorder is-findMatch is-hovered"
  cx={50}
  cy={50}
  r={49}
  vectorEffect="non-scaling-stroke"
/>
`;

exports[`getNodeRenderers vectorBorder returns circle with .is-findMatch,	without .is-hovered,	with .is-pathHovered,	and with .is-focalNode 1`] = `
<circle
  className="DdgNode--VectorBorder is-findMatch is-pathHovered is-focalNode"
  cx={50}
  cy={50}
  r={49}
  vectorEffect="non-scaling-stroke"
/>
`;

exports[`getNodeRenderers vectorBorder returns circle with .is-findMatch,	without .is-hovered,	with .is-pathHovered,	and without .is-focalNode 1`] = `
<circle
  className="DdgNode--VectorBorder is-findMatch is-pathHovered"
  cx={50}
  cy={50}
  r={49}
  vectorEffect="non-scaling-stroke"
/>
`;

exports[`getNodeRenderers vectorBorder returns circle with .is-findMatch,	without .is-hovered,	without .is-pathHovered,	and with .is-focalNode 1`] = `
<circle
  className="DdgNode--VectorBorder is-findMatch is-focalNode"
  cx={50}
  cy={50}
  r={49}
  vectorEffect="non-scaling-stroke"
/>
`;

exports[`getNodeRenderers vectorBorder returns circle with .is-findMatch,	without .is-hovered,	without .is-pathHovered,	and without .is-focalNode 1`] = `
<circle
  className="DdgNode--VectorBorder is-findMatch"
  cx={50}
  cy={50}
  r={49}
  vectorEffect="non-scaling-stroke"
/>
`;

exports[`getNodeRenderers vectorBorder returns circle without .is-findMatch,	with .is-hovered,	with .is-pathHovered,	and with .is-focalNode 1`] = `
<circle
  className="DdgNode--VectorBorder is-hovered is-pathHovered is-focalNode"
  cx={50}
  cy={50}
  r={49}
  vectorEffect="non-scaling-stroke"
/>
`;

exports[`getNodeRenderers vectorBorder returns circle without .is-findMatch,	with .is-hovered,	with .is-pathHovered,	and without .is-focalNode 1`] = `
<circle
  className="DdgNode--VectorBorder is-hovered is-pathHovered"
  cx={50}
  cy={50}
  r={49}
  vectorEffect="non-scaling-stroke"
/>
`;

exports[`getNodeRenderers vectorBorder returns circle without .is-findMatch,	with .is-hovered,	without .is-pathHovered,	and with .is-focalNode 1`] = `
<circle
  className="DdgNode--VectorBorder is-hovered is-focalNode"
  cx={50}
  cy={50}
  r={49}
  vectorEffect="non-scaling-stroke"
/>
`;

exports[`getNodeRenderers vectorBorder returns circle without .is-findMatch,	with .is-hovered,	without .is-pathHovered,	and without .is-focalNode 1`] = `
<circle
  className="DdgNode--VectorBorder is-hovered"
  cx={50}
  cy={50}
  r={49}
  vectorEffect="non-scaling-stroke"
/>
`;

exports[`getNodeRenderers vectorBorder returns circle without .is-findMatch,	without .is-hovered,	with .is-pathHovered,	and with .is-focalNode 1`] = `
<circle
  className="DdgNode--VectorBorder is-pathHovered is-focalNode"
  cx={50}
  cy={50}
  r={49}
  vectorEffect="non-scaling-stroke"
/>
`;

exports[`getNodeRenderers vectorBorder returns circle without .is-findMatch,	without .is-hovered,	with .is-pathHovered,	and without .is-focalNode 1`] = `
<circle
  className="DdgNode--VectorBorder is-pathHovered"
  cx={50}
  cy={50}
  r={49}
  vectorEffect="non-scaling-stroke"
/>
`;

exports[`getNodeRenderers vectorBorder returns circle without .is-findMatch,	without .is-hovered,	without .is-pathHovered,	and with .is-focalNode 1`] = `
<circle
  className="DdgNode--VectorBorder is-focalNode"
  cx={50}
  cy={50}
  r={49}
  vectorEffect="non-scaling-stroke"
/>
`;

exports[`getNodeRenderers vectorBorder returns circle without .is-findMatch,	without .is-hovered,	without .is-pathHovered,	and without .is-focalNode 1`] = `
<circle
  className="DdgNode--VectorBorder"
  cx={50}
  cy={50}
  r={49}
  vectorEffect="non-scaling-stroke"
/>
`;

exports[`getNodeRenderers vectorFindColorBand returns circle with correct size and className 1`] = `
<circle
  className="DdgNode--VectorFindEmphasis--colorBand"
  cx={50}
  cy={50}
  r={49}
  vectorEffect="non-scaling-stroke"
/>
`;
